### ISSUE TITLE
Deno 1.31.0 sometimes fails importing non-statically analyzable dynamic imports concurrently

### ISSUE THREAD

**Issue by oscarotero** [denoland/deno#17918]
Hi.
I've starting to get this weird error on Deno 1.31.0:
https://github.com/lumeland/lume/actions/runs/4262008619/jobs/7417000682#step:7:1139

It works fine on 1.30.3. I've tried to replicate this error in a isolated environment without lucky.
Maybe you have some clue why it happens?

```
Caused by TypeError: Loading unprepared module: file:///home/runner/work/lume/lume/tests/assets/jsx_preact/with-function.jsx#1677238255917, imported from: file:///home/runner/work/lume/lume/core/loaders/module.ts
```

**Comment by dsherret** [comment]
I have also not been able to create a simple example reproducing this, but I have been able to when running:

```
deno test -A --no-check -L debug --filter "build a site with jsx/tsx modules using Preact" tests/jsx_preact.test.ts
```

...in Lume's codebase. I am looking into it.

**Comment by dsherret** [comment]
Found the issue. This may occurs when loading two or more non-statically analyzable dynamic imports at the same time... silly mistake I did. Will submit a fix and add a test.

**Comment by oscarotero** [comment]
Great! Thanks for the quick reply and fix!

**Comment by dsherret** [comment]
@oscarotero should be fixed in 1.31.1 (now published)

**Comment by oscarotero** [comment]
It works like a charm. Thanks!
